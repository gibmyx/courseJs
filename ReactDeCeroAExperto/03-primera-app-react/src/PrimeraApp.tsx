import React from "react";

const PrimeraApp = ({saludo, subtitulo = 'Soy un subtitulo'}: { saludo:string }) => {

    return (
        <>
            <h1>{saludo}</h1>
            {/*<pre>{JSON.stringify(saludo, null, 3)}</pre>*/}
            <p>{subtitulo}</p>
        </>
    )
}

export default PrimeraApp