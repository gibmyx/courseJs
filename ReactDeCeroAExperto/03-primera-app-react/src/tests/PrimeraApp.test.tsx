import React from "react";
import {describe, test, expect} from "vitest";

import PrimeraApp from "../PrimeraApp";
import {shallow} from "enzyme";

describe('Prueba en <PrimeraApp />', () => {
    test('debe mostrar el mensaje de Hola, soy Goku', () => {
        const saludo = "Hola, soy Goku";
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>)
        expect(wrapper).toMatchSnapshot()
    })

    test('debe mostrar el subtitulo enviado por props', () => {
        const saludo = "Hola, soy Goku";
        const subtitulo = "Soy un subtitulo";
        const wrapper = shallow(<PrimeraApp
            saludo={saludo}
            subtitulo={subtitulo}
        />)

        const textoParrafo = wrapper.find('p').text()
        expect(textoParrafo).toBe(subtitulo)
    })
})
