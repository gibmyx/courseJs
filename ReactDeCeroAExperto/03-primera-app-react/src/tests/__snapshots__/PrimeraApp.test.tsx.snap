// Vitest Snapshot v1

exports[`Prueba en <PrimeraApp /> > debe mostrar el mensaje de Hola, soy Goku 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;

exports[`Prueba en <PrimeraApp /> > debe mostrar el subtitulo enviado por props 1`] = `
<Fragment>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
